.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,140 @@
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        required: true
+        type: string
+      gochannel:
+        required: true
+        type: string
+      unit-scenario:
+        required: true
+        type: string
+
+jobs:
+  unit-tests:
+    runs-on: ${{ inputs.runs-on }}
+    env:
+      GOPATH: ${{ github.workspace }}
+      # Set PATH to ignore the load of magic binaries from /usr/local/bin And
+      # to use the go snap automatically. Note that we install go from the
+      # snap in a step below. Without this we get the GitHub-controlled latest
+      # version of go.
+      PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
+      GOROOT: ""
+      
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        # needed for git commit history
+        fetch-depth: 0
+        # NOTE: checkout the code in a fixed location, even for forks, as this
```

Comment:
How about I add another task for myself after this PR is merged to do some cleanup? Since the file changes make reviewing this PR difficult, I think perhaps it would be better to have only reorganization in this PR and actual changes in another so that it's easy for reviewers to see the changes.

Reply:
I suppose this is no longer an issue since we added go.mod support. Perhaps it's worth to check if we could simply use the project default location. FWIW, followup material.